raiden/raiden_event_handler.py

```diff
@@ -331,16 +331,16 @@ def handle_paymentsentfailed(
         payment_status = raiden.targets_to_identifiers_to_statuses[target].pop(
             payment_identifier, None
         )
-
-        handle_receive_events_with_payments(raiden.wal.storage,
-                                            payment_status.payment_hash_invoice,
-                                            'raiden.transfer.events.EventPaymentSentFailed',
-                                            payment_identifier)
+        if payment_status:
```

Comment:
Why is this needed? If payment status is not present then we have a bug here.

Reply:
handle_receive_events_with_payments is making use of payment_status. So the condition will always be true. 
In the case that payment_status if None the function handle_receive_events_with_payments will make the node to fall. In the case that we use to fix the refund, a LockedExpired is making the node fall because of this line.

Reply:
I understand. I think we should fix this in a future. Basically we have a function that is invoked every time that a payment fail, and there are some situations where the `payment_status` exists and in others dont. We should split this into different functions.

Reply:
Agree we should do that